common.go

```diff
@@ -18,89 +18,83 @@ package ledger_cosmos_go
 
 import (
 	"encoding/binary"
+	"errors"
 	"fmt"
+	"strconv"
+	"strings"
 )
 
-// VersionInfo contains app version information
-type VersionInfo struct {
-	AppMode uint8
-	Major   uint8
-	Minor   uint8
-	Patch   uint8
+type VersionResponse struct {
+	AppMode   uint8 // 0: release | 0xFF: debug
+	Major     uint8
+	Minor     uint8
+	Patch     uint8
+	AppLocked uint8
+	TargetId  uint32
 }
 
-func (c VersionInfo) String() string {
-	return fmt.Sprintf("%d.%d.%d", c.Major, c.Minor, c.Patch)
-}
-
-// VersionRequiredError the command is not supported by this app
-type VersionRequiredError struct {
-	Found    VersionInfo
-	Required VersionInfo
+type AddressResponse struct {
+	pubkey  []byte // 0: release | 0xFF: debug
+	address string
 }
 
-func (e VersionRequiredError) Error() string {
-	return fmt.Sprintf("App Version required %s - Version found: %s", e.Required, e.Found)
+type SignatureResponse struct {
+	signatureDER []byte
 }
 
-func NewVersionRequiredError(req VersionInfo, ver VersionInfo) error {
-	return &VersionRequiredError{
-		Found:    ver,
-		Required: req,
-	}
+func (c VersionResponse) String() string {
+	return fmt.Sprintf("%d.%d.%d", c.Major, c.Minor, c.Patch)
 }
 
-// CheckVersion compares the current version with the required version
-func CheckVersion(ver VersionInfo, req VersionInfo) error {
-	if ver.Major != req.Major {
-		if ver.Major > req.Major {
-			return nil
-		}
-		return NewVersionRequiredError(req, ver)
+// Validate HRP: Max length = 83
+// All characters must be in range [33, 126], displayable chars in Ledger devices
+func serializeHRP(hrp string) (hrpBytes []byte, err error) {
+	if len(hrp) > 83 {
```

Comment:
Avoid using magic numbers if possible, for example here you could create a `constants.go` file and add something like:
```go
const (
HRP_MAX_LENGTH = 83
// all other constant values
)
```